src/Elastic.Apm/Report/PayloadSenderV2.cs

```diff
@@ -91,6 +92,13 @@ internal class PayloadSenderV2 : IPayloadSender, IDisposable
 			string AdaptUserAgentValue(string value) => Regex.Replace(value, "[ /()<>@,:;={}?\\[\\]\"\\\\]", "_");
 		}
 
+		private static HttpClientHandler CreateHttpClientHandler(IConfigurationReader configurationReader) =>
+			new HttpClientHandler
+			{
+				ServerCertificateCustomValidationCallback = (message, certificate, chain, policyError) =>
+					policyError == SslPolicyErrors.None || !configurationReader.VerifyServerCert
```

Comment:
Call `configurationReader.VerifyServerCert` cause re-parsing of the option - I'm working on fixing this performance issue in another PR but for this PR I think it would be better to just cache the value for example as one of the lambda captures    .

Reply:
++

I thought there may be some desire to make configuration changes reloadable, based on https://github.com/elastic/apm-agent-dotnet/blob/9efee030ac820e66767c71d5e61948d6c188c38f/src/Elastic.Apm.AspNetCore/Config/MicrosoftExtensionsConfig.cs#L47, but can pass the `bool` configuration value directly to `CreateHttpClientHandler()`

Reply:
Yes, we will make it re-loadable but from performance point of view the right way to do it is by re-parsing configuration only on change and not on each read access. Do you agree? (BTW this what I plan to do in that other PR that I mentioned above - I'll add a note to myself to make sure we read option value from the latest configuration snapshot in this callback).